tools/wasi/wasi.cpp

```diff
@@ -0,0 +1,235 @@
+    {"wasi_snapshot_preview1", "proc_exit", {fizzy::ValType::i32}, std::nullopt, wasi_proc_exit},
+    // clock_time_get
+    // fd_read
+    {"wasi_snapshot_preview1", "fd_read",
+        {fizzy::ValType::i32, fizzy::ValType::i32, fizzy::ValType::i32, fizzy::ValType::i32},
+        fizzy::ValType::i32, wasi_fd_read},
+    {"wasi_snapshot_preview1", "fd_write",
+        {fizzy::ValType::i32, fizzy::ValType::i32, fizzy::ValType::i32, fizzy::ValType::i32},
+        fizzy::ValType::i32, wasi_fd_write},
+    {"wasi_snapshot_preview1", "fd_prestat_get", {fizzy::ValType::i32, fizzy::ValType::i32},
+        fizzy::ValType::i32, wasi_fd_prestat_get},
+    {"wasi_snapshot_preview1", "fd_prestat_dir_name",
+        {fizzy::ValType::i32, fizzy::ValType::i32, fizzy::ValType::i32}, fizzy::ValType::i32,
+        wasi_return_enosys},
+    {"wasi_snapshot_preview1", "environ_sizes_get", {fizzy::ValType::i32, fizzy::ValType::i32},
+        fizzy::ValType::i32, wasi_environ_sizes_get},
+    {"wasi_snapshot_preview1", "environ_get", {fizzy::ValType::i32, fizzy::ValType::i32},
+        fizzy::ValType::i32, wasi_return_enosys},
+
+
+    {"wasi_unstable", "proc_exit", {fizzy::ValType::i32}, std::nullopt, wasi_proc_exit},
+    // clock_time_get
+    // fd_read
+    {"wasi_unstable", "clock_time_get",
+        {fizzy::ValType::i32, fizzy::ValType::i64, fizzy::ValType::i32}, fizzy::ValType::i32,
+        wasi_return_enosys},
+    {"wasi_unstable", "fd_read",
+        {fizzy::ValType::i32, fizzy::ValType::i32, fizzy::ValType::i32, fizzy::ValType::i32},
+        fizzy::ValType::i32, wasi_fd_read},
+    {"wasi_unstable", "fd_write",
+        {fizzy::ValType::i32, fizzy::ValType::i32, fizzy::ValType::i32, fizzy::ValType::i32},
+        fizzy::ValType::i32, wasi_fd_write},
+    {"wasi_unstable", "fd_prestat_get", {fizzy::ValType::i32, fizzy::ValType::i32},
+        fizzy::ValType::i32, wasi_fd_prestat_get},
+    {"wasi_unstable", "fd_prestat_dir_name",
+        {fizzy::ValType::i32, fizzy::ValType::i32, fizzy::ValType::i32}, fizzy::ValType::i32,
+        wasi_return_enosys},
+    {"wasi_unstable", "environ_sizes_get", {fizzy::ValType::i32, fizzy::ValType::i32},
+        fizzy::ValType::i32, wasi_environ_sizes_get},
+    {"wasi_unstable", "environ_get", {fizzy::ValType::i32, fizzy::ValType::i32},
+        fizzy::ValType::i32, wasi_return_enosys},
+};
+
+bool run(int argc, char** argv)
+{
+    uvwasi_options_t options = {
+        3,           // sizeof fd_table
+        0, nullptr,  // NOTE: no remappings
+        static_cast<uvwasi_size_t>(argc), const_cast<const char**>(argv),
+        nullptr,  // TODO: support env
+        0, 1, 2,
+        nullptr,  // NOTE: no special allocator
+    };
+
+    uvwasi_errno_t err = uvwasi_init(&state, &options);
+    if (err != UVWASI_ESUCCESS)
+    {
+        std::cerr << "Failed to initialise UVWASI: " << uvwasi_embedder_err_code_to_string(err)
+                  << std::endl;
```

Comment:
```suggestion
                  << "\n";
```

Reply:
I think in these cases the flushing `std::endl` does is beneficial/ok. Any other reason not to use it?

Reply:
Only for consistency. stderr is unbuffered anyway.